amqpstorm/tests/unit/message_tests.py

```diff
@@ -44,6 +44,27 @@ def test_message_default_properties(self):
         self.assertIsInstance(message.correlation_id, str)
         self.assertIsInstance(message.timestamp, datetime)
 
+    def test_message_create_does_not_mutate_properties(self):
+        properties = {
+            'content_type': 'application/json',
+            'headers': {'key': 'value'}
+        }
+
+        message1 = Message.create(None, self.message, properties)
+        message2 = Message.create(None, self.message, properties)
+
+        self.assertIsInstance(message1.message_id, str)
+        self.assertIsInstance(message1.correlation_id, str)
+        self.assertIsInstance(message1.timestamp, datetime)
+
+        self.assertIsInstance(message2.message_id, str)
+        self.assertIsInstance(message2.correlation_id, str)
+        self.assertIsInstance(message2.timestamp, datetime)
+
+        self.assertNotEqual(message1.message_id, message2.message_id)
+        self.assertNotEqual(message1.correlation_id, message2.correlation_id)
+        self.assertNotEqual(message1.timestamp, message2.timestamp)
+
```

Comment:
Great test!
Would it be worthwhile to also explicitly check for the absence of the keys in properties?
```
self.assertNotIn('correlation_id', properties)
self.assertNotIn('message_id', properties)
self.assertNotIn('timestamp', properties)
```

Or alternatively to make a copy of properties before calling Message.create and asserting that properties and its unused copy are identical afterwards.

Reply:
Thanks! Incorporated your feedback into the PR.